cmd/queue/main.go

```diff
@@ -202,33 +203,33 @@ func proxyHandler(reqChan chan queue.ReqEvent, breaker *queue.Breaker, tracingEn
 	}
 }
 
-func preferPodForScaledown(logger *zap.SugaredLogger, downwardAPILabelsPath string) bool {
+func preferPodForScaledown(downwardAPILabelsPath string) (bool, error) {
 	// Short circuit a rejection when no label path file is mounted
 	if _, err := os.Stat(downwardAPILabelsPath); os.IsNotExist(err) {
-		return false
+		return false, nil
 	}
```

Comment:
return the err?

Reply:
We shouldnt here. Its effectively a check if the feature is enabled.

Reply:
I would ignore the errors returned from this method in general.

Reply:
+1 on logging and ignoring the errors. the error shouldn't decide the behavior, the bool should.

Reply:
now since we're only logging it, should we return the error here? It's probably an interesting error, if `Stat` failed.

Reply:
Yeah I agree we should return error here now

Reply:
We should only return this error if it's not a `IsNotExist` error. Again, this is a short-circuit path that tells us that the feature is disabled. We absolutely don't want error logs in the queue-proxy just because the preferScaleDown feature is disabled.